runner: Avoid local network collissions

sarjona opened the pull request:
When running it locally, a couple of issues have been identified:

- There are collisions if there are other local networks called `moodle*`. The filter for searching the NETWORKNAME has been changed to fix it.
- The owner for the /var/www folder has been also changed to www-data for the PHPUnit tests.

stronk7 commented on runner/master/run.sh:
```diff
@@ -165,7 +165,7 @@ echo
 echo "============================================================================"
 NETWORKNAME="moodle"
-NETWORK=$(docker network list -q --filter name="${NETWORKNAME}")
+NETWORK=$(docker network list -q --filter name="${NETWORKNAME}$")
```
While I like it (really didn't know that PCRE is available there)... I was just guessing if, maybe,  we could make the ```NETWORKNAME``` parametrizable, still defaulting to "moodle", but allowing to create/use different ones. We already do that for other env variables. Sort of:

````
NETWORKNAME ="${NETWORKNAME:-moodle}"
````

Other than that, open to discussion. The patch looks perfect!

andrewnicols commented:
Why do we need to change the owner for the /var/www directory? We do so for behat but there shouldn't be any need for phpunit since we execute phpunit as root anyway.

sarjona commented:
@stronk7 thanks for the suggestion! I've added the NETWORKNAME also as parameter following your indications!

@andrewnicols you're right! Thanks a lot for reviewing it!! Yesterday I wasn't able to execute PHPUnit without that change, but today everything is working as expected. I've removed that part of the patch!

Ready again to review! :-)

stronk7 commented:
Sold!

BTW, couple of comments:

1) note that all these changes are auto-deployed @ ci.moodle.org (every 5 mins, from memory). nightlysetup (salt does it for us).

2) also note that there are 2 branches, master and testing. When we want to test something in place (@ ci.moodle.org)... we put our experiments in the testing branch, and then you can tell to any job (see the DEV ones) to use the "TESTING" branch instead of the "STABLE" one. Sometimes it's useful. Also, from time to time, when nobody is using "testing" we reset it back to "master". (testing can be rewritten, master cannot).

3) About 2)... we still have to find a better way to handle testing stuff (other repos don't have that testing/stable duality). But right now, that's what we have.

Ciao :-)
